### ISSUE TITLE
[BUG] Chromium does not launch on macos 11.2.1 without explicit `executablePath`

### ISSUE THREAD

**Issue by patrickberkeley** [microsoft/playwright#5561]
**Context:**
- Playwright Version: 1.8.1
- Operating System: macos 11.2.1 Intel
- Node.js version: 14.15.5
- Browser: Chromium
- Extra: <none>

Running a spec from the command line using:

```sh
DEBUG=pw:api PWDEBUG=1 npx folio tests/<test>.spec.ts -p browserName=chromium
```

Results in in the error:

```sh
browserType.launch: No executable path is specified. Pass "executablePath" option directly.
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
```

This seems similar to the error in: https://github.com/microsoft/playwright/issues/4722. And I only started seeing it after upgrading macos to 11.2.1.

**Comment by patrickberkeley** [comment]
Here's the npm error log:

```
0 verbose cli [
0 verbose cli   '/Users/patrickberkeley/.volta/tools/image/node/14.15.5/bin/node',
0 verbose cli   '/Users/patrickberkeley/.volta/tools/image/npm/7.5.4/bin/npm-cli.js',
0 verbose cli   'exec',
0 verbose cli   '--',
0 verbose cli   'folio',
0 verbose cli   'tests/some-spec.spec.ts',
0 verbose cli   '-p',
0 verbose cli   'browserName=chromium'
0 verbose cli ]
1 info using npm@7.5.4
2 info using node@v14.15.5
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/Users/patrickberkeley/.volta/tools/image/npm/7.5.4/npmrc Completed in 0ms
5 timing config:load:builtin Completed in 0ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/Users/patrickberkeley/Code/pa/chimney/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/Users/patrickberkeley/.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:/Users/patrickberkeley/.volta/tools/image/node/14.15.5/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 5ms
19 verbose npm-session 1f349bb03318d0c3
20 timing npm:load Completed in 10ms
21 timing command:exec Completed in 1976ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/Users/patrickberkeley/.volta/tools/image/npm/7.5.4/node_modules/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:315:20)
22 verbose stack     at maybeClose (internal/child_process.js:1048:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
23 verbose pkgid chimney@1.0.0
24 verbose cwd /Users/patrickberkeley/Code/pa/chimney
25 verbose Darwin 20.3.0
26 verbose argv "/Users/patrickberkeley/.volta/tools/image/node/14.15.5/bin/node" "/Users/patrickberkeley/.volta/tools/image/npm/7.5.4/bin/npm-cli.js" "exec" "--" "folio" "tests/some-spec.spec.ts" "-p" "browserName=chromium"
27 verbose node v14.15.5
28 verbose npm  v7.5.4
29 error code 1
30 error path /Users/patrickberkeley/Code/pa/chimney
31 error command failed
32 error command sh -c folio "tests/some-spec.spec.ts" "-p" "browserName=chromium"
33 verbose exit 1
```

**Comment by patrickberkeley** [comment]
I can get test running again if I do explicitly specify the [`exectutablePath`](https://playwright.dev/docs/api/class-browsertype?_highlight=executablepath#browsertypeexecutablepath) option:

```ts
import { folio as baseFolio } from '@playwright/test';
import { LaunchOptions } from 'playwright';
import { Config } from '@helpers/config';

type TestFixtures = {
  config: Config,
};

const builder = baseFolio.extend<TestFixtures>();

builder.browserOptions.override(async ({ browserOptions }, runTest) => {
  const modifiedOptions: LaunchOptions = {
    ...browserOptions, // default

    executablePath: '/Applications/Chromium.app/Contents/MacOS/Chromium',
  }

  await runTest(modifiedOptions);
});
```

So it seems just the packaged version of chromium is not found.

**Comment by aslushnikov** [comment]
@patrickberkeley could you please try with `playwright@next`? We rewrote this part, so it might be fixed on tip-of-tree?

**Comment by aslushnikov** [comment]
Also, are on ARM or Intel macbook?

**Comment by aslushnikov** [comment]
For the record: we found a Mac 11.2.1 macbook and tried there and didn't manage to reproduce this. Thus removing the P1 label.

**Comment by patrickberkeley** [comment]
@aslushnikov I'm running with an intel chip. And yes! It does work with `playwright@next` (`1.9.1-1614187757000`). 

As a side note, and in case anyone else runs into it, I also install `playwright-test` so I can do global overrides from `folio`. `playwright-test` installs an older version of `playwright` (highest version is `1.8.0` I believe). I know I'm not telling you anything new since you have [a PR to bump playwright on playwright-test](https://github.com/microsoft/playwright-test/pull/217). I tried installing from you playwright-test branch, but that doesn't work due to how the package files are moved into place with the prePublish hook. The only way I could try out next was forcing it with a yarn resolution.

**Comment by aslushnikov** [comment]
@patrickberkeley the playwright-test will be updated once the playwright v1.9.1 will be released - should be very soon. 

And thank you for verifying that things work!

**Comment by patrickberkeley** [comment]
✅ closing since there's a clear path forward. Thanks @aslushnikov.

**Comment by CharlieMartell** [comment]
We are still seeing this issue on @playwright/test@1.12.3. @patrickberkeley were you able to solve the issue by upgrading?

**Comment by patrickberkeley** [comment]
@CharlieMartell yeah the issue was resolved for me in playwrite v1.9.1
